File: Backend/demo/src/main/java/com/example/Models/InvoiceHeader.java
package com.example.Models;

import java.time.LocalDate;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.*;

@Entity
public class InvoiceHeader {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long invoiceId;

    @ManyToOne
    @JoinColumn(name = "bookingId", nullable = false)
    @JsonBackReference 
    private BookingHeader booking;
    
    private String cName; 
    private String cEmailId; 
    private String cMobileNo; 
    private String cAadharNo; 
    private String cPassNo; 

    @ManyToOne
    @JoinColumn(name = "customerId", nullable = false)
    private CustomerMaster customer;
    
    @OneToMany(mappedBy = "invoice", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<InvoiceDetails> invoiceDetails;
    
    private Long pickup_hubId;
    private Long return_hubId;

    @Enumerated(EnumType.STRING)
    @Column(name = "isReturned", length = 1)
    private Return_Status isReturned;
    
    public enum Return_Status {
        Y, N
    }

    private LocalDate handoverDate;

    @ManyToOne
    @JoinColumn(name = "carId", nullable = false)
    private CarMaster car;

    private LocalDate returnDate;
    private double rentalAmt;
    private double totalAddOnAmt;
    private double totalAmt;
    private String rate;

    // Getters and Setters
    public Long getInvoiceId() {
        return invoiceId;
    }

    public void setInvoiceId(Long invoiceId) {
        this.invoiceId = invoiceId;
    }

    public BookingHeader getBooking() {
        return booking;
    }

    public void setBooking(BookingHeader booking) {
        this.booking = booking;
    }

    public CustomerMaster getCustomer() {
        return customer;
    }

    public void setCustomer(CustomerMaster customer) {
        this.customer = customer;
    }

    public LocalDate getHandoverDate() {
        return handoverDate;
    }

    public void setHandoverDate(LocalDate handoverDate) {
        this.handoverDate = handoverDate;
    }

    public CarMaster getCar() {
        return car;
    }

    public void setCar(CarMaster car) {
        this.car = car;
    }

    public LocalDate getReturnDate() {
        return returnDate;
    }

    public void setReturnDate(LocalDate returnDate) {
        this.returnDate = returnDate;
    }

    public double getRentalAmt() {
        return rentalAmt;
    }

    public void setRentalAmt(double rentalAmt) {
        this.rentalAmt = rentalAmt;
    }

    public double getTotalAddOnAmt() {
        return totalAddOnAmt;
    }

    public void setTotalAddOnAmt(double totalAddOnAmt) {
        this.totalAddOnAmt = totalAddOnAmt;
    }

    public double getTotalAmt() {
        return totalAmt;
    }

    public void setTotalAmt(double totalAmt) {
        this.totalAmt = totalAmt;
    }

    public String getRate() {
        return rate;
    }

    public void setRate(String rate) {
        this.rate = rate;
    }

    public Return_Status getIsReturned() {
        return isReturned;
    }

    public void setIsReturned(Return_Status isReturned) {
        this.isReturned = isReturned;
    }

    public Long getPickup_hubId() {
        return pickup_hubId;
    }

    public void setPickup_hubId(Long pickup_hubId) {
        this.pickup_hubId = pickup_hubId;
    }

    public Long getReturn_hubId() {
        return return_hubId;
    }

    public void setReturn_hubId(Long return_hubId) {
        this.return_hubId = return_hubId;
    }

    @Override
    public String toString() {
        return "InvoiceHeader [invoiceId=" + invoiceId + ", booking=" + booking + ", cName=" + cName + 
                ", cEmailId=" + cEmailId + ", cMobileNo=" + cMobileNo + ", cAadharNo=" + cAadharNo + 
                ", cPassNo=" + cPassNo + ", customer=" + customer + ", pickup_hubId=" + pickup_hubId + 
                ", return_hubId=" + return_hubId + ", isReturned=" + isReturned + ", handoverDate=" + handoverDate + 
                ", car=" + car + ", returnDate=" + returnDate + ", rentalAmt=" + rentalAmt + 
                ", totalAddOnAmt=" + totalAddOnAmt + ", totalAmt=" + totalAmt + ", rate=" + rate + "]";
    }
}

File: Dot net Backend/latest/Fleet_Management/Models/InvoiceHeader.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Fleet_Management.Models;

[Table("invoice_header")]
[Index("BookingId", Name = "FK1rk1jp3v1teuk31m3jpg4ug2i")]
[Index("CarId", Name = "FKd2t74dthekdhhrfb3pottooy4")]
[Index("CustomerId", Name = "FKhky9vtepslur0k09ifs19qkp3")]
public partial class InvoiceHeader
{
    [Key]
    [Column("invoice_id")]
    public long InvoiceId { get; set; }

    [Column("c_aadhar_no")]
    [StringLength(255)]
    public string? CAadharNo { get; set; }

    [Column("c_email_id")]
    [StringLength(255)]
    public string? CEmailId { get; set; }

    [Column("c_mobile_no")]
    [StringLength(255)]
    public string? CMobileNo { get; set; }

    [Column("c_name")]
    [StringLength(255)]
    public string? CName { get; set; }

    [Column("c_pass_no")]
    [StringLength(255)]
    public string? CPassNo { get; set; }

    [Column("handover_date")]
    public DateOnly? HandoverDate { get; set; }

    [Column("is_returned", TypeName = "enum('N','Y')")]
    public string? IsReturned { get; set; }

    [Column("pickup_hub_id")]
    public long? PickupHubId { get; set; }

    [Column("rate")]
    [StringLength(255)]
    public string? Rate { get; set; }

    [Column("rental_amt")]
    public double RentalAmt { get; set; }

    [Column("return_date")]
    public DateOnly? ReturnDate { get; set; }

    [Column("return_hub_id")]
    public long? ReturnHubId { get; set; }

    [Column("total_add_on_amt")]
    public double TotalAddOnAmt { get; set; }

    [Column("total_amt")]
    public double TotalAmt { get; set; }

    [Column("booking_id")]
    public long BookingId { get; set; }

    [Column("car_id")]
    public long CarId { get; set; }

    [Column("customer_id")]
    public long CustomerId { get; set; }

    [ForeignKey("BookingId")]
    [InverseProperty("InvoiceHeaders")]
    public virtual Booking Booking { get; set; } = null!;

    [ForeignKey("CarId")]
    [InverseProperty("InvoiceHeaders")]
    public virtual CarMaster Car { get; set; } = null!;

    [ForeignKey("CustomerId")]
    [InverseProperty("InvoiceHeaders")]
    public virtual CustomerMaster Customer { get; set; } = null!;

    [InverseProperty("Invoice")]
    public virtual ICollection<InvoiceDetail> InvoiceDetails { get; set; } = new List<InvoiceDetail>();
}
